test/inductor/test_cpu_repro.py

```diff
@@ -1599,6 +1600,33 @@ def f(a):
         x = torch.rand(2, 3, 14, 14).bfloat16()
         self.common(f, (x,))
 
+    def test_eliminate_meaningless_copy(self):
+        def fn(x1, x2):
+            permute = torch.ops.aten.permute.default(x2, [0, 2, 1, 3])
+            clone = torch.ops.aten.clone.default(
+                permute, memory_format=torch.contiguous_format
+            )
+            view = torch.ops.aten.view.default(clone, [1024, -1, 32])
+            bmm = torch.ops.aten.bmm.default(view, x1)
+            permute = torch.ops.aten.permute.default(view, [0, 2, 1])
+            return (bmm, permute)
+
+        inps = [
+            rand_strided(
+                (1024, 32, 128), (4096, 1, 32), device="cpu", dtype=torch.float32
+            ),
+            rand_strided(
+                (64, 128, 16, 32),
+                (65536, 512, 32, 1),
+                device="cpu",
+                dtype=torch.float32,
+            ),
+        ]
+        fn_opt = torch._dynamo.optimize("inductor")(fn)
+        code = run_and_get_cpp_code(fn_opt, *inps)
+        self.assertTrue("cpp_fused_permute_1" not in code)
```

Comment:
this seems fragile, kernel naming is changed frequently. Do you want to count the number of kernels instead?

Reply:
Updated.